doc/api/next_api_changes/behaviour.rst

```diff
@@ -147,3 +147,11 @@ The parameter ``s`` to `.Axes.annotate` and  `.pyplot.annotate` is renamed to
 The old parameter name remains supported, but
 support for it will be dropped in a future Matplotlib release.
 
+``get_cmap()`` now returns a copy of the colormap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Previously, calling ``get_cmap()`` would return
+the built-in Colormap. If you made modifications to that colormap, the
+changes would be propagated in the global state. This function now
+returns a copy of all registered colormaps to keep the built-in
```

Comment:
"... copy of the requested colormap..."?

Reply:
Should this even go into this now if it is a future change and there are not currently any behavior changes?

My thoughts are to actually break these commits up into two separate PRs. One with the deprecation warning/preparation for a future version. Then another one with the proposed copy code in it that could be pushed off for quite a while and discussed for immutability or not.

Reply:
I personally think the globals should be immutable.  If folks are modifying the global to then call again as a named colormap, then I think they were using a misfeature/bug.  

I just gave a comment here because I was reading it, and had a quick thought.  Please work out w/ Tim Hoffman how to do the rest of the PR, but this seems to be the right direction to be heading.